### PR TITLE
Types Bundler: Add package

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:

This PR adds a new package to plugin-tools called `@grafana/plugin-types-bundler` which will allow plugin devs to create `index.d.ts` files for their plugins to share with other plugins. The main use case here is for sharing plugin extension related types.

The package itself marries grafana/levitate with dts-bundle-generator to bundle and treeshake all types (including imported types from other packages) into a single file.

**Which issue(s) this PR fixes**:

Related grafana/grafana/issues/90171

**Special notes for your reviewer**:

This is the same (types-bundler package code wise) as #1015 which I've had to close due to bizarre package-lock.json issues that I cannot figure out.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@5.3.11-canary.1092.a678194.0
  npm install @grafana/plugin-e2e@1.8.1-canary.1092.a678194.0
  npm install @grafana/plugin-meta-extractor@0.0.7-canary.1092.a678194.0
  npm install @grafana/plugin-types-bundler@0.0.1-canary.1092.a678194.0
  npm install @grafana/sign-plugin@3.0.4-canary.1092.a678194.0
  # or 
  yarn add @grafana/create-plugin@5.3.11-canary.1092.a678194.0
  yarn add @grafana/plugin-e2e@1.8.1-canary.1092.a678194.0
  yarn add @grafana/plugin-meta-extractor@0.0.7-canary.1092.a678194.0
  yarn add @grafana/plugin-types-bundler@0.0.1-canary.1092.a678194.0
  yarn add @grafana/sign-plugin@3.0.4-canary.1092.a678194.0
  ```
</details>

### CODE REVIEW

**Comment by academo** [comment]
Would you consider adding some integration tests in levitate to make sure this case is supported as we update levitate too?

**Comment by jackw** [comment]
> Would you consider adding some integration tests in levitate to make sure this case is supported as we update levitate too?

Integration tests for levitates `getImportsInfo` function? Yeah I can do that. I'll likely be adding some tests for this package too as probably to assert it's generating what we expect when other deps update.

File: packages/plugin-types-bundler/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export type ParsedArgs = {
+  entryPoint?: string;
+  tsConfig: string;
+  outDir: string;
+};
+
+export const parsedArgs = parseArgs<ParsedArgs>(args, {
+  alias: {
+    entryPoint: 'entry-point',
+    tsConfig: 'ts-config',
```
Comment (sunker) [inline packages/plugin-types-bundler/src/utils.ts:15]: Should this arg be called project to aling with tsc? I'm not sure.

File: packages/plugin-types-bundler/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Usage
+
+#### CLI
+
+By calling the binary against an (module.ts|tsx) file, the package will output a single index.d.ts file in the dist directory with your other plugin assets.
+
+```bash
+# Run local build
+./dist/run.js ~/my-grafana-plugin/src/types/index.ts
```
Comment (sunker) [inline packages/plugin-types-bundler/README.md:22]:
```suggestion
./dist/run.js --entry-point ~/my-grafana-plugin/src/types/index.ts
```

File: packages/plugin-types-bundler/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# Run latest remote version
+npx @grafana/plugin-types-bundler ~/my-grafana-plugin/src/types/index.ts
```
Comment (sunker) [inline packages/plugin-types-bundler/README.md:25]:
```suggestion
npx @grafana/plugin-types-bundler --entry-point ~/my-grafana-plugin/src/types/index.ts
```

**Review by sunker — COMMENTED** [review]
👏  

I tried bundling the slo app and the asserts app locally, but faced different problems when running the bundler script. Am I missing something? Do consumers need to change something on their side for bundling to work?

File: packages/plugin-types-bundler/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+```
+npm install @grafana/plugin-types-bundler --save-dev
+```
+
+## Usage
+
+#### CLI
+
+By calling the binary against an (module.ts|tsx) file, the package will output a single index.d.ts file in the dist directory with your other plugin assets.
```
Comment (leventebalogh) [inline packages/plugin-types-bundler/README.md:18]: Hmm, should we change the `(module.ts|tsx)` bit to something like `(types.ts|tsx)`, as we are not planning to run it against the plugin's module file?

File: packages/plugin-types-bundler/src/bundleTypes.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import { getImportsInfo } from '@grafana/levitate';
+import { EntryPointConfig, generateDtsBundle } from 'jackw-dts-bundle-gen-test';
+import { writeFileSync } from 'node:fs';
+import { extname, join } from 'node:path';
+import { parsedArgs } from './utils.js';
+
+export const generateTypes = () => {
+  const { entryPoint, tsConfig, outDir } = parsedArgs;
+
+  if (!entryPoint) {
```
Comment (leventebalogh) [inline packages/plugin-types-bundler/src/bundleTypes.ts:10]: **Nit:** although it has no harm, should we check it here as well? (We are already validating it in the `run.ts`.)

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Comment by leventebalogh** [comment]
I pushed two commits, feel free to override / remove them @jackw:
- fixed an error that happens when the `dist/` directory isn't yet created (edge-case)
- added a min node version, mostly because [one of the node APIs used by levitate](https://nodejs.org/api/events.html#eventsaddabortlistenersignal-listener) doesn't below `v18.18.0`

**Review by sunker — APPROVED** [review]
Tested and it works as expected. Great work on this guys 🎉

**Pull request merged**